lib/smb2-cmd-write.c

```diff
@@ -262,10 +262,11 @@ smb2_process_write_request_fixed(struct smb2_context *smb2,
         smb2_get_uint32(iov, 4, &req->length);
         smb2_get_uint64(iov, 8, &req->offset);
         memcpy(req->file_id, iov->buf + 16, SMB2_FD_SIZE);
-        smb2_get_uint32(iov, 24, &req->channel);
-        smb2_get_uint32(iov, 28, &req->remaining_bytes);
-        smb2_get_uint16(iov, 32, &req->write_channel_info_offset);
-        smb2_get_uint16(iov, 34, &req->write_channel_info_length);
+        smb2_get_uint32(iov, 32, &req->channel);
```

Comment:
The offsets for write_channel_info_offset, info_length and flags seems wrong. The look to be 2 too high. Same above.

Can you check this?

Reply:
Great catch.. they were even more wrong before this change which kind of worked, but now they should be correct